src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -79,6 +83,36 @@ def gen_proposal_reply(self, best_proposer: str, prp_orders: dict, messages: Mes
                     proposer, str(msg)
                 )
         return messages
+    
+    def respond_to_invalid_orders(self, invalid_proposal_orders: Dict[str, List[str]], messages_data: MessagesData) -> None:
+        """
+        The bot responds by HUHing the invalid proposal orders received (this could occur if the move proposed is invalid)
+
+        :param invalid_proposal_orders: dictionary of sender -> invalid orders proposed
+        :param messages_data: Message Data object to add messages
+        """
+        if not invalid_proposal_orders:
+            return
+        for sender in invalid_proposal_orders:
+            message = HUH(PRP(ORR(XDO(dipnet_to_daide_parsing(invalid_proposal_orders[sender])))))
+            messages_data.add_message(
+                sender, str(message)
+            )
+        
+    def respond_to_alliance_messages(self, messages_data: MessagesData) -> None:
+        """
+        Send YES confirmation messages to all alliance proposals
+        :param messages_data: Message Data object to add messages
+        """
+        unique_senders = {}
+        for sender, message in self.alliances.values():
+            unique_senders[sender] = message
+        for sender, message in unique_senders.items():
+            messages_data.add_message(sender, str(YES(message)))
+
+        if self.alliances:
+            print("Alliances accepted")
```

Comment:
Because we haven't yet tested any of this functionality. So, I was intending to keep such print statements for now so that while testing we can see that they are working. I was thinking of removing it in a later PR.